public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -414,6 +415,36 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     return this._initialState;
   }
 
+  public duplicateLibraryPanel(libraryPanel: LibraryVizPanel) {
+    if (!libraryPanel.parent) {
+      return;
+    }
+    const gridItem = libraryPanel.parent;
+
+    if (!(gridItem instanceof SceneGridItem || PanelRepeaterGridItem)) {
+      console.error('Trying to duplicate a panel in a layout that is not SceneGridItem or PanelRepeaterGridItem');
+      return;
+    }
```

Comment:
this condition is the same as in duplicatePanel, we should combine this method and duplicatePanel to share this initial logic. Would be nice if panel menu could just call dashboard.duplicatePanel. I do not like that the code that calls duplicatePanel needs to care about if it's a library panel or not